pagefaultgames/pokerogue
# [P3][UI] Remove unnecessary sprite reloads in Starter Select UI

## What are the changes the user will see?
Pokemon sprites in Starter Select will no longer refresh when cycling through nature, ability, or when changing gender if it makes no difference in the sprite


## Why am I making these changes?
It looks bad and is unnecessary ([discord bug report](https://discord.com/channels/1125469663833370665/1289790514899583128))




## What are the changes from a developer perspective?
- Check the attributes that are getting updated in `setSpeciesDetails` and only update the sprite if form, shiny/variant or gender are changed
- removed 2 unused class attributes
- somewhat cleaned up the Utils imports




### Screenshots/Videos
Before: Sprites blink and get their animation reset when cycling through nature, gender, etc

https://github.com/user-attachments/assets/92c06c14-13df-4d91-8f69-97221bf67285


After: 

https://github.com/user-attachments/assets/fbad76fc-da63-48f5-a011-8f6fc230c025





## How to test the changes?
Start a new game and cycle through various starters and characteristics. The sprites should only update when necessary





## Checklist
